x-pack/plugins/canvas/public/components/workpad_loader/index.tsx

```diff
@@ -0,0 +1,167 @@
+        services.notify.error(err, { title: errors.getFindFailureErrorMessage() });
+      }
+    },
+    [services.notify, services.workpad]
+  );
+
+  const onDownloadWorkpad = useCallback((workpadId: string) => downloadWorkpad(workpadId), []);
+
+  const cloneWorkpad = useCallback(
+    async (workpadId: string) => {
+      try {
+        const workpad = await services.workpad.get(workpadId);
+        workpad.name = strings.getClonedWorkpadName(workpad.name);
+        workpad.id = getId('workpad');
+        await services.workpad.create(workpad);
+        history.push(`/workpad/${workpad.id}/page/1`);
+      } catch (err) {
+        services.notify.error(err, { title: errors.getCloneFailureErrorMessage() });
+      }
+    },
+    [services.notify, services.workpad, history]
+  );
+
+  const removeWorkpads = useCallback(
+    (workpadIds: string[]) => {
+      if (workpadsState === null) {
+        return;
+      }
+
+      const removedWorkpads = workpadIds.map(async (id) => {
+        try {
+          await services.workpad.remove(id);
+          return { id, err: null };
+        } catch (err) {
+          return { id, err };
+        }
+      });
+
+      return Promise.all(removedWorkpads).then((results) => {
+        let redirectHome = false;
+
+        const [passes, errored] = results.reduce<[string[], string[]]>(
+          ([passesArr, errorsArr], result) => {
+            if (result.id === fromState.workpadId && !result.err) {
+              redirectHome = true;
+            }
+
+            if (result.err) {
+              errorsArr.push(result.id);
+            } else {
+              passesArr.push(result.id);
+            }
+
+            return [passesArr, errorsArr];
+          },
+          [[], []]
+        );
+
+        const remainingWorkpads = workpadsState.workpads.filter(({ id }) => !passes.includes(id));
+
+        const workpadState = {
+          total: remainingWorkpads.length,
+          workpads: remainingWorkpads,
+        };
+
+        if (errored.length > 0) {
+          services.notify.error(errors.getDeleteFailureErrorMessage());
+        }
+
+        setWorkpadsState(workpadState);
+
+        if (redirectHome) {
+          history.push('/');
+        }
+
+        return errored;
+      });
+    },
+    [history, services.workpad, fromState.workpadId, workpadsState, services.notify]
+  );
+
+  const formatDate = useCallback(
+    (date: any) => {
+      const dateFormat = services.platform.getUISetting('dateFormat');
+      return date && moment(date).format(dateFormat);
+    },
+    [services.platform]
+  );
+
+  return (
+    <Component
+      workpadId={fromState.workpadId}
+      canUserWrite={fromState.canUserWrite}
+      cloneWorkpad={cloneWorkpad}
+      createWorkpad={createWorkpad}
+      findWorkpads={findWorkpads}
+      downloadWorkpad={onDownloadWorkpad}
+      removeWorkpads={removeWorkpads}
+      onClose={onClose}
+      workpads={workpadsState}
+      formatDate={formatDate}
```

Comment:
This is easier as:

```
const { workpadId, canUserWrite } = fromState;
<Component
  downloadWorkpad={onDownloadWorkpad}
  workpads={workpadsState}
  {...{
    workpadId,
    canUserWrite,
    cloneWorkpad,
    createWorkpad,
    findWorkpads,
    removeWorkpads, 
	onClose,
	formatDate,
  }}
/>
```

I'd love to see consistency in event handlers, (e.g. `downloadWorkpad` becomes `onDownloadWorkpad`, `createWorkpad` becomes `onCreateWorkpad`, etc)